app/src/main/java/com/example/capstone/LoginFragment.java

```diff
@@ -180,13 +169,40 @@ public void onComplete(@NonNull Task<AuthResult> task) {
                             // Sign in success, update UI with the signed-in user's information
                             Log.d(TAG, "signInWithCredential:success");
                             FirebaseUser user = mAuth.getCurrentUser();
-                            updateUI(user);
 
-                            // TODO: comeback later to cleanup updateUI
-                            Intent home = new Intent(getActivity(), HomescreenActivity.class);
-                            startActivity(home);
+                            // update parse db
+                            User registeredUser = new User();
+                            String firebase_uid = user.getUid();
+
+                            // check if user already exists
```

Comment:
A few things here:
1. Use clear variable names even if they are verbose:
a. There are two users here: `user` and `registeredUser`. It's unclear what they represent just by reading their names. Better names are `firebaseUser` and `parseUser`. But `parseUser` should not be created on line 174, but rather before line 184 as it is only conditionally used.
b. `firebase_uid` is used on line 180 but not on line 184, why? The variable is not needed IMO, just use `firebaseUser. getUid()`.

2. Break code blocks into individual methods rather than using comments:
`isNewUser(String uid)`
`registerNewUser(String uid)`
`navigateToProfileCreation()`
`navigateToHome()`

3. Use consistent naming:
I missed this last time, but this file uses both `signIn` and `logIn`.